### ISSUE TITLE
Inclusion of United States census supplementary questions

### ISSUE THREAD

**Issue by Ziris85** [fisharebest/webtrees#3258]
In the 1940 (and upcoming 1950) United States censuses, there are at the bottom some supplementary questions included that only apply to specific people as they are added to the census. I've had a couple, but found that the census assistant doesn't include them. Given the unlikely probability of having very many of these to include in your own records, it doesn't seem practical to include them by default in the default assistant code since that would include a ton of empty/blank entries.

After doing some experimenting with shared note formatting, it looks like multiple tables can be included in the same note, such as:
```
.start_formatted_area.
.b.Name|.b.Relation
Bob Smith|Self
.end_formatted_area.

.start_formatted_area.
.b.Nickname|.b.Personality
Robert|Pretty wacky
.end_formatted_area.
```
I was wondering if it might be possible to include an additional toggle to include these supplementary questions in the shared note in a similar fashion? I was thinking how, when the census date is selected from the drop-down and the "Create a shared note using the census assistant" link appears, for those censuses that have supplementary questions, include an additional link that would pull that in, such as "Include Supplementary Questions?". When clicked, a new area would be shown below the normal census data which would include those questions.

I hope that makes sense. Thoughts?

**Comment by fisharebest** [comment]
Given the design of the census system, I think the only option would be to create a second copy of the 1940 census with additional columns.  So, you'd choose from a list like:

```
1920
1930
1940
1940 (with supplementary)
1950
1950 (with supplementary)
```

**Comment by Ziris85** [comment]
I like that idea better lol. I'd be happy to write up the necessary files where I'm able, but I'm wondering if the code allows for what I mentioned earlier - creating multiple tables? Usually with those supplementary questions you're dealing with only one individual, and trying to include all of those questions into a single row will make for a VERY long row heh...just trying to avoid creating a row that scrolls a mile past the edge of my screen.

**Comment by fisharebest** [comment]
> Usually with those supplementary questions you're dealing with only one individual

OK - I have never seen one of these censuses.

But we already have a free-text notes/comments field at the end of the table.

Might it make more sense to use that field for these supplementary questions, and add the answers as simple text?

**Comment by Ziris85** [comment]
So, from what I can see after looking a bit further, the supplementary questions were only asked on the 1940 & 1950 censuses, so probably not worth the hassle of trying to include special functionality just for these two. (side-note, the 1960 census looks like a beast lol, so that'll be interesting trying to incorporate that....in like 12 years lol...) However, I do like the formatting that the table provides, so not sure if the free text notes area would be appropriate there.

I suppose if you have some documentation area for this, I just wrote up a proposal formatting for the two censuses that I suppose folks could just append manually to their special notes for the few cases where they have relatives who answered those questions:
**1940**
```
.start_formatted_area.
.b.Name|.b.FBP|.b.MBP|.b.MotherLang|.b.Veteran|.b.VetFather|.b.Service|.b.SSN|.b.Deductions|.b.DeductAmt|.b.UsualOccupation|.b.UsualIndustry|.b.UsualClass|.b.NumMarr|.b.AM|.b.Children
.end_formatted_area.
```
**1950**
```
.start_formatted_area.
.b.SameHouse|.b.Farm|.b.SameCounty|.b.County|.b.State|.b.FBP|.b.MBP|.b.Grade|.b.Finished|.b.School|.b.WeeksLooking|.b.WeeksWorking|.b.Income|.b.Business|.b.OtherIncome|.b.RelIncome|.b.RelBusiness|.b.RelOther|.b.WWII|.b.WWI|.b.AnyWar
.end_formatted_area.
```

**Comment by fisharebest** [comment]
The markup using `.b.` is from webtrees 1.
webtrees 2 uses markdown to create tables.

You should use markdown.

If we convert the comments field from a single-line into a multi-line, then users can add markdown tables (or anything else) there.

**Comment by Ziris85** [comment]
Oh nice! I didn't know that - definitely good to know, thanks! And yeah, agreed that if markdown will be rendered in that comments section, then that would be just as good a place as any for it. Also, re-wrote that stuff in markdown:

1940
```
|Name|FBP|MBP|MotherLang|Veteran|VetFather|Service|SSN|Deductions|DeductAmt|UsualOccupation|UsualIndustry|UsualClass|NumMarr|AM|Children|
|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
```

1950
```
|SameHouse|Farm|SameCounty|County|State|FBP|MBP|Grade|Finished|School|WeeksLooking|WeeksWorking|Income|Business|OtherIncome|RelIncome|RelBusiness|RelOther|WWII|WWI|AnyWar|
|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
```

**Comment by fisharebest** [comment]
OK - I'll make the comments entry multi-line.

That will also fix #3236

**Comment by miqrogroove** [comment]
This sounds nice and will save me from some extra work too.  Since I started using webtrees for editing my tree, I wanted to customize the census fields and decided the easiest way was to make my own shared note and then copy/paste the raw gedcom for the census event to each person manually.  Being able to add my own multiline table in the census assistant would be great.
